optihood/MPC/interface.py

```diff
@@ -254,24 +256,50 @@ def get_mpc_scenario_from_csv(self, input_folder_path: _pl.Path) -> dict[str, di
 
         return system_state
 
-    def update_nodal_data(self, current_state: dict[str, dict[str, float]]) -> dict:
+    def update_nodal_data(self, current_state: dict[str, dict[str, float]]
+                          ) -> dict:
         """Requires self.nodal_data"""
         # TODO: clip profiles?
         # TODO: adjust building model parameters to correct implementation.
         nodal_data = _cp.deepcopy(self.nodal_data)
         for label, inputs in current_state.items():
             sheet_name = self.label_to_sheet[label]
             sheet = nodal_data[sheet_name]
-            label_column = _ent.CommonLabels.label_unique
-            if sheet_name == _ent.NodeKeysOptional.building_model_parameters:
-                label_column = _ent.BuildingModelParameters.building_unique
+            label_column = ent.CommonLabels.label_unique
+
+            if sheet_name == ent.NodeKeysOptional.building_model_parameters:
+                label_column = ent.BuildingModelParameters.building_unique
+
             row_index = sheet[label_column] == label
             for column_name, value in inputs.items():
                 # TODO: deal with incompatible dtype. Sometimes int instead of float.
                 sheet.loc[row_index, column_name] = value
 
         return nodal_data
 
+    @staticmethod
+    def clip_profiles(nodal_data: dict, current_time_period: _pd.DatetimeIndex):
+        """Should only be applied on the copy of the initial nodal_data."""
+        profs = ent.ProfileTypes
+
+        # each building has its own df of demand profiles
+        dfs = nodal_data[profs.demandProfiles]
+        for key, df in dfs.items():
+            dfs[key] = re.ProfileAndOtherDataReader.clip_to_time_index(df, current_time_period)
+
+        mandatory_profile_sheets = [profs.weather]  # excluding demand profiles.
+        for sheet in mandatory_profile_sheets:
+            df = nodal_data[sheet]
+            nodal_data[sheet] = re.ProfileAndOtherDataReader.clip_to_time_index(df, current_time_period)
+
+        non_mandatory_profile_sheets = [profs.internal_gains, profs.fixed_sources, profs.electricity_impact, profs.electricity_cost]
```

Comment:
I think we should add the `mandatory_profile_sheets` and `non_mandatory_profile_sheets` to entities.